test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -455,10 +455,20 @@ func TestTierTemplateRevision(t *testing.T) {
 			// then
 			// an additional TTR will be created
 			// TODO check for exact match or remove the +1 once we implement the cleanup controller
-			ttrsWithNewParams, err := hostAwait.WaitForTTRs(t, customTier.Name, wait.GreaterOrEqual(len(updatedTTRs)+1))
+			ttrsWithNewParams, err := hostAwait.WaitForTTRs(t, customTier.Name, wait.GreaterOrEqual(2*len(ttrs)+1))
 			require.NoError(t, err)
 			// retrieve new tier once the ttrs were created and the revision field updated
-			customTier.NSTemplateTier, err = hostAwait.WaitForNSTemplateTier(t, customTier.Name)
+			customTier.NSTemplateTier, err = wait.For(t, hostAwait.Awaitility, &toolchainv1alpha1.NSTemplateTier{}).
+				WithNameMatching(customTier.Name, func(actual *toolchainv1alpha1.NSTemplateTier) bool {
+					for _, oldValue := range updatedCustomTier.Status.Revisions {
+						for _, actualValue := range actual.Status.Revisions {
```

Comment:
until we merge the cleanup logic, not all revisions will be replaced.

Reply:
good point. It will update the entries in the revisions , you're right! Sorry for the confusion!